lightning/src/ln/channelmanager.rs

```diff
@@ -5051,44 +5050,86 @@ where
 											&self.secp_ctx, b.inbound_blinding_point, &encrypted_tlvs_ss
 										).ok()
 									});
-									if let Err(e) = chan.queue_add_htlc(outgoing_amt_msat,
-										payment_hash, outgoing_cltv_value, htlc_source.clone(),
-										onion_packet, skimmed_fee_msat, next_blinding_point, &self.fee_estimator,
-										&&logger)
-									{
-										if let ChannelError::Ignore(msg) = e {
-											log_trace!(logger, "Failed to forward HTLC with payment_hash {}: {}", &payment_hash, msg);
+
+									// Forward the HTLC over the most appropriate channel with the corresponding peer,
+									// applying non-strict forwarding.
+									// The channel with the least amount of outbound liquidity will be used to maximize the
+									// probability of being able to successfully forward a subsequent HTLC.
+									let mut channels_with_peer = peer_state.channel_by_id.values_mut().filter_map(|phase| match phase {
+										ChannelPhase::Funded(chan) => Some(chan),
+										_ => None,
+									}).collect::<Vec<&mut Channel<_>>>();
```

Comment:
Done. This was mainly done to avoid duplication with the validation in `send_htlc`. Do we also want to verify the establishment/shutdown state? https://github.com/lightningdevkit/rust-lightning/blob/737df0f46f3fcbd8fb4abf4cf17330f09927893c/lightning/src/ln/channel.rs#L7049-L7083

Reply:
Hmm, the last commit triggered a disconnection in the fuzzer, is this potentially due to calling the fee estimator in `get_available_balances`?

Reply:
> Do we also want to verify the establishment/shutdown state?

To check back in here, I did add an `is_usable()` check which I think is necessary, but let me know if that's not the case.